airflow/providers/amazon/aws/utils/exponential_backoff_retry.py

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import logging
+from datetime import datetime, timedelta
+from typing import Callable
+
+from airflow.utils import timezone
+
+log = logging.getLogger(__name__)
+
+
+def exponential_backoff_retry(
```

Comment:
We use retry backoff as part of BaseOperator code
https://github.com/apache/airflow/blob/f60d458dc08a5d5fbe5903fffca8f7b03009f49a/airflow/models/taskinstance.py#L1983

It's not quite the same but I wonder if we want to maintain two backoff retry logics?

Reply:
I think the differences between the two are enough that we can have them as separate functions. The retry logic in the core code is dependent on variables tied to the task instance, which makes it difficult to apply to generic objects. This function also has the concept of the `callable_function` which is going to be difficult to adapt to the core use-case. I think it is better to be kept as a separate function - otherwise we will introduce a lot of complexities for not a whole lot of benefit.